R/internal.R

```diff
@@ -258,7 +258,7 @@ check_response <- function(resp, query = NULL) {
 
 is_id <- function(.) {
   out <- tryCatch({
-    .out <- all(grepl("[[:alnum:]]{8}\\-[[:alnum:]]{4}\\-[[:alnum:]]{4}\\-[[:alnum:]]{4}\\-[[:alnum:]]{12}", .)) && all(!is.na(.)) && !is.null(.) && is.character(.)
+    .out <- all(grepl("^[[:alnum:]]{8}\\-[[:alnum:]]{4}\\-[[:alnum:]]{4}\\-[[:alnum:]]{4}\\-[[:alnum:]]{12}$", .)) && all(!is.na(.)) && !is.null(.) && is.character(.)

```

Comment:
Hi @ndrewGele ,
Thanks for the PR!
Identifying strings that fit the template for an ID but might be longer is actually by design. I was anticipating that people will append a suffix to existing order IDs as a custom order ID to track related orders. This is actually demonstrated in the vignette on `order_id`. 
Does this compromise something you're trying to do? 
Perhaps you can provide a logical argument like `strict = FALSE` (please make this the default so it doesn't break) to the function and add your regex conditionally when `strict  = TRUE`?

Reply:
Gotcha. I think that's fine if that's how you want `is_id()` to work. 
The issue I experienced was because the existing logic in `orders()` won't accept client order ID's that don't follow the same template format as Alpaca's order ID's. 

For example:
`AlpacaforR::orders(symbol_id = 'asdfgh123456', status = 'all', client_order_id = TRUE)` 
Returns:
```
Error: code: 422
message: client_order_id is missing
```
(because `.o_id` is set to `NULL` when the existing logic thinks 'asdfgh123456" must be a stock ticker since it doesn't look like an Alpaca Order ID.)

I think rearranging the logic I wrote for `orders()` will fix my issue even if I revert the changes here to `is_id()`.

Reply:
Ahhh, I see. Noted - I will take it into account that `symbol_id` (or `.o_id` as it were)  needs to be passed downstream to the `get_url` regardless of condition when `client_order_id = TRUE` to avoid reintroducing this bug.